docs/v0.9/en/user-guide/operations/compaction.md

```diff
@@ -0,0 +1,134 @@
+# Compaction
+
+For databases based on the LSM Tree, compaction is extremely critical. It merges overlapping fragmented SST files into a single ordered file, discards deleted data while significantly improves query performance.
+
+Until v0.9.1, GreptimeDB provides compaction strategies to control how SST files are compacted: Time windowed compaction strategy (TWCS) and strict window compaction strategy.
+
+
+## Concepts
+
+Let us start with those core concepts of compaction in GreptimeDB.
+
+### SST files
+
+SSTs are sorted files generated when memtables are flushed to persistent storage, such as disks and object stores.
+
+In GreptimeDB, data rows in SST files are organized by primary keys and timestamps, as illustrated below. Each SST file covers a specific time range. When a query specifies a time range, GreptimeDB retrieves only the relevant SST files that may contain data within that range, rather than loading all persisted files.
+
+![SST layout](/compaction-sst-file-layout.jpg)
+
+Typically, the time ranges of SST files do not overlap during real-time writing workloads. However, due to factors like row deletions and out-of-order writes, SST files may have overlapping time ranges, which can affect query performance.
+
+### Time window
+
+Time-series workloads present prominent "windowed" pattern in that most recently inserted rows are more possible to be read. Thus GreptimeDB logically partitioned the time axis into different time windows and we are more interested in compacting those SST files fall into the same time window. 
+
+Time window for some specific table is usually inferred from the most-recently flushed SST files or users may manually specify the time window if TWCS is chosen.
+
+GreptimeDB has a preset collection of window sizes that are:
+- 1 hour
+- 2 hours
+- 12 hours
+- 1 day
+- 1 week
+- 1 year
+- 10 years
+
+If time window is not specified, GreptimeDB will infer it when the first compaction happens by selecting the minimum time window size for the above collection that can cover the whole time span of all files to be compacted.
+
+For example, during the first compaction, the time span for all SST files is 4 hours, then GreptimeDB will choose 12 hours as the time window for that table and persist it for later compactions.
+
+GreptimeDB considers the window contains most recently inserted timestamps as **active window** and those previous windows as **inactive windows**.
+
+### Sorted runs
+Sorted runs is a collection of SST files that have sorted and non-overlapping time ranges. 
+
+For example, a table contains 5 SSTs with following time ranges (all inclusive): `[0, 10]`, `[12, 23]`, `[22, 25]`,`[24, 30]`,`[26,33]` and we can find 2 sorted runs:
```

Comment:
For each array, we should maintain a consistent format with either no spaces or one space between elements.
```suggestion
For example, a table contains 5 SSTs with following time ranges (all inclusive): `[0, 10]`, `[12,23]`, `[22,25]`,`[24,30]`,`[26,33]` and we can find 2 sorted runs:
```